nft-marketplace/src/offers.rs

```diff
@@ -3,13 +3,14 @@ use gstd::{exec, msg, prelude::*, ActorId};
 use market_io::*;
 use primitive_types::{H256, U256};
 
-fn get_hash(nft_contract_id: &ActorId, ft_contract_id: Option<ActorId>, price: u128) -> H256 {
+fn get_hash(nft_contract_id: &ActorId, token_id: U256, ft_contract_id: Option<ActorId>, price: u128) -> H256 {
     let nft_conract_vec: Vec<u8> = <[u8; 32]>::from(*nft_contract_id).into();
     let price_vec: Vec<u8> = price.to_be_bytes().into();
+    let token_id_vec: Vec<u8> = <[u8; 32]>::from(token_id).into();
     let ft_contract_vec: Vec<u8> = ft_contract_id
         .map(|id| <[u8; 32]>::from(id).into())
         .unwrap_or_default();
-    sp_core_hashing::blake2_256(&[nft_conract_vec, price_vec, ft_contract_vec].concat()).into()
+    sp_core_hashing::blake2_256(&[nft_conract_vec, price_vec, ft_contract_vec, token_id_vec].concat()).into()
```

Comment:
I'm not sure it is more optimal, but it will produce fewer allocations (arrays are stored on the stack, `Vec`s on the heap). Need to benchmark!

```rust
    let nft_id = <[u8; 32]>::from(*nft_contract_id);
    let price = price.to_be_bytes();
    let token_id = <[u8; 32]>::from(token_id);
    let ft_id_vec: Vec<u8> = ft_contract_id
        .map(|id| <[u8; 32]>::from(id).into())
        .unwrap_or_default();
    sp_core_hashing::blake2_256(&[&nft_id[..], &price[..], &ft_id_vec[..], &token_id[..]].concat())
        .into()
```